frontend/src/Pages/PostPage.tsx

```diff
@@ -59,6 +63,11 @@ export default function PostPage() {
         if (location.hash) {
             commentId = parseInt(location.hash.substring(1));
             scrollToComment = document.querySelector<HTMLDivElement>(`[data-comment-id="${commentId}"]`);
+
+            if (!scrollToComment && fetchingComment !== commentId) {
+                reload(unreadOnly).then(() => setFetchingComment(commentId));
```

Comment:
I think both cases can be fixed by relying on the last read timestamp/comment instead of the materialized "isNew" flag.

This allows performing the comment presence check easily, as well as loading the "new" comments without removing existing "new" comments from the page.